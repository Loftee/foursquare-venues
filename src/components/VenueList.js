import React from 'react';

import './VenueList.scss';

function VenueList(props) {
    const {venues, selectVenue} = props;
    return (
        <div className={'venue-list__wrapper'}>
            <h2>Venue List</h2>
            <ul className={'venue-list'}>
                {
                    venues.map(venue => {
                        return (
                            <li key={venue.id} className={'venue-list__item'} onClick={() => {
                                selectVenue(venue.id)
                            }}>
                                <h3>{venue.name}</h3>
                                <p>{venue.location.distance} meters away</p>
                            </li>
                        )
                    })
                }
            </ul>
        </div>
    )
}

export default VenueList;