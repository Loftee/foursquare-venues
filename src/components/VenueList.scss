.venue-list {
  width: 100%;
  list-style: none;
  padding: 0;
  @media (max-width: 767px) {
    display: flex;
    flex-wrap: nowrap;
    overflow-x: auto;
    overflow-y: hidden;
  }

  &__wrapper {
    width: 25%;
    padding: 0 15px;
    float: left;
    @media (max-width: 767px) {
      width: 100%;
    }
  }

  &__item {
    margin-bottom: 20px;
    padding: 10px;
    border: 1px solid #fff;
    cursor: pointer;
    transition: padding 0.2s linear;
    border-radius: 5px;
    @media (max-width: 767px) {
      margin-left: 10px;
      min-width: 40%;
    }
  }

  &__item:hover {
    padding-left: 15px;
    padding-right: 5px;
  }

}