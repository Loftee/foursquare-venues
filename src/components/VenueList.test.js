import React from 'react';
import {shallow} from 'enzyme';
import VenueList from './VenueList';

describe('VenueList component', () => {
    const props = {
        venues: [{
            "id": "4c3d6ed94bc9a5939b07d471",
            "name": "Costa Coffee",
            "location": {
                "address": "4 Great Portland St",
                "lat": 51.51573111967288,
                "lng": -0.14068655684295295,
                "labeledLatLngs": [{"label": "display", "lat": 51.51573111967288, "lng": -0.14068655684295295}],
                "distance": 42,
                "postalCode": "W1W 8QJ",
                "cc": "GB",
                "city": "London",
                "state": "Greater London",
                "country": "United Kingdom",
                "formattedAddress": ["4 Great Portland St", "London", "Greater London", "W1W 8QJ", "United Kingdom"]
            }
        }]
    };
    const component = shallow(<VenueList {...props} />);
    it('displays a list of venues', () => {
        expect(component.find('.venue-list__item').length).toEqual(1);
    });
});