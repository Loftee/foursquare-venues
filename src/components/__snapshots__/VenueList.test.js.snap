// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VenueList component renders correctly 1`] = `
<div
  className="venue-list__wrapper"
>
  <h2>
    Venue List
  </h2>
  <ul
    className="venue-list"
  >
    <li
      className="venue-list__item"
      key="4c3d6ed94bc9a5939b07d471"
    >
      <h3>
        Costa Coffee
      </h3>
      <p>
        4 Great Portland St
      </p>
    </li>
  </ul>
</div>
`;
